dsl_parser/functions.py

```diff
@@ -79,6 +85,19 @@ def _convert_attribute_list_to_python_syntax_string(attr_list):
 _register_entry_point_functions()
 
 
+def _contains_legal_nested_attribute_path_items(l):
+    return all(map(
+        lambda x:
+        _is_function(x) or isinstance(x, (basestring, int)),
+        l))
```

Comment:
can be written nicer as `return all(_is_function(x) or isinstance(x, (basestring, int)) for x in l)` without the map+lambda